contrib/librdkafka-cmake/CMakeLists.txt

```diff
@@ -114,8 +118,11 @@ endif()
 
 add_library(_rdkafka ${SRCS})
 add_library(ch_contrib::rdkafka ALIAS _rdkafka)
-
 target_compile_options(_rdkafka PRIVATE -fno-sanitize=undefined)
+
+add_library(_rdkafka_cjson ${RDKAFKA_SOURCE_DIR}/cJSON.c)
+target_compile_definitions(_rdkafka_cjson PRIVATE -DCJSON_HIDE_SYMBOLS)
```

Comment:
Let's add a comment that w/o `-DCJSON_HIDE_SYMBOLS` we collide with aws libraries.

Reply:
that 

> > Hm. It [looks like](https://s3.amazonaws.com/clickhouse-test-reports/43259/c01cc4790fa01d67e6affb2923ef14aac7ed85ce/stateful_tests__tsan_/stderr.log) aws-s3 still links with cJSON.c
> > ```
> > WARNING: ThreadSanitizer: data race (pid=702)
> >   Write of size 8 at 0x00002789d2c0 by thread T370:
> >     #0 cJSON_ParseWithLengthOpts build_docker/../contrib/librdkafka/src/cJSON.c:1018:31 (clickhouse+0x221b885e) (BuildId: 1bc55adc0c9724fbd4e2861ae91875cd725c324a)
> >     #1 cJSON_ParseWithOpts build_docker/../contrib/librdkafka/src/cJSON.c:1004:16 (clickhouse+0x221b950e) (BuildId: 1bc55adc0c9724fbd4e2861ae91875cd725c324a)
> >     #2 cJSON_Parse build_docker/../contrib/librdkafka/src/cJSON.c:1086:16 (clickhouse+0x221b950e)
> >     #3 aws_json_value_new_from_string <null> (clickhouse+0x229f4d0b) (BuildId: 1bc55adc0c9724fbd4e2861ae91875cd725c324a)
> >     #4 aws_endpoints_path_through_object <null> (clickhouse+0x22b40184) (BuildId: 1bc55adc0c9724fbd4e2861ae91875cd725c324a)
> >     #5 s_resolve_template endpoints_rule_engine.c (clickhouse+0x22b4239e) (BuildId: 1bc55adc0c9724fbd4e2861ae91875cd725c324a)
> >     #6 aws_byte_buf_init_from_resolved_templated_string <null> (clickhouse+0x22b4a652) (BuildId: 1bc55adc0c9724fbd4e2861ae91875cd725c324a)
> >     #7 s_resolve_expr endpoints_rule_engine.c (clickhouse+0x22b3fafd) (BuildId: 1bc55adc0c9724fbd4e2861ae91875cd725c324a)
> >     #8 aws_endpoints_rule_engine_resolve <null> (clickhouse+0x22b41c85) (BuildId: 1bc55adc0c9724fbd4e2861ae91875cd725c324a)
> > ```
> 
> @filimonov With reference to your comment [#43259 (comment)](https://github.com/ClickHouse/ClickHouse/pull/43259#issuecomment-1403598829) ... Yes, the AWS libraries in "contrib/" set `CJSON_HIDE_SYMBOLS` BUT the CMake files which your search found are not the bundled CMake file which the ClickHouse build calls. This will be "contrib/aws-cmake/CMakeLists.txt" which does NOT set `CJSON_HIDE_SYMBOLS`. Maybe try if the error goes away if `CJSON_HIDE_SYMBOLS` is set in there?

But that seems the problem is the opposite -aws now founds the lib from librdkafka (despite all PRIVATE / CJSON_HIDE_SYMBOLS etc). I think I'm giving up and will just try a silly hacky workaround - let librdkafka link with cjson from aws s3, unless aws s3 is disabled. Let's see if it will help...